### DISCUSSION TITLE
Making some useful stuff with {worldtime}

### DISCUSSION THREAD

**Discussion by ElectraBytes04** [NGSpace/hudder#10] [General]
# I wanted to make a hud that would tell the player when they could sleep.

### Here's what I have so far:
```
&9FPS: &%fps<30, "c", fps<60, "e", "r"%{fps}
%worldtime>=12542, {cansleep=true}, worldtime<=12541, {cansleep=false}%
Time - Ticks: {worldtime} %{cansleep=true}, "- Sleep!", {cansleep=false}, ""%
```
How would I make the variables in line 2 not show up?
## Other things I am trying to do:
Make worldtime display as a regular time format, not just ticks. In HUDDER code!
So far:
```
{worldtime}
{seconds=worldtime/20}
{minutes=seconds/60}
{hours=minutes/60}
```
I haven't done any formatting yet.

**Comment by ElectraBytes04** [comment]
Is there a way to check for multiples of an integer variable in a condition? Like this:
```
%{worldtime>=12542(multiples)}, {cansleep=true}, {worldtime>=2400(multiples)}, {cansleep=false}, {cansleep=false}%
```
Maybe finishing this in JavaScript

**Reply by NGSpace** [reply]
I don't really understand what you're asking  but if you're asking how to multiply numbers with variables you can
```
%worldtime>=12542*multiples, {cansleep=true}, worldtime>=2400*multiples, {cansleep=false}, {cansleep=false}%
```

**Reply by ElectraBytes04** [reply]
Turns out I didn't need to check multiples of any numbers. Also, I ended up switching to JavaScript.
```javascript
function topleft() {
	let time;

	if (getVal("worldtime") > 24000) {
    	time = getVal("worldtime") - 24000 * Math.floor(getVal("worldtime") / 24000);
	} else {
    	time = getVal("worldtime");
	}
	
	let hour, minute;
	
	if ((Math.floor(time / 1000) + 6) > 23) {
	    hour = (Math.floor(time / 1000) + 6) - 24;
	} else {
	    hour = (Math.floor(time / 1000) + 6);
	}
	
	minute = Math.floor((time * 60) / 1000) % 60;
	
	let cansleep = false;
	
	if (hour >= 18 && minute >= 32) {
		cansleep = true;
	}

	if (cansleep === false) {
		return "\u00A79Time: \u00A7r" + hour + ":" + (minute < 10 ? "0" + minute : minute);
	} else if (cansleep === true) {
		return "\u00A79Time: \u00A7r" + hour + ":" + (minute < 10 ? "0" + minute : minute) + " - Sleep!";
	}
}
```

**Reply by NGSpace** [reply]
JavaScript is definitely one of the most useful features I added lol

Although in recent versions the default hudder compiler has really gone up in quality. It will be even better when [this PR](https://github.com/NGSpace/hudder/pull/8#issue-2419620261) will get finalized and merged.

**Reply by NGSpace** [reply]
BTW may I add that script into hudder as one of it's builtin huds? there is kind of a lack of JS default scripts...

**Reply by ElectraBytes04** [reply]
Oh yea sure! Sorry, I didn't see your comment.

**Comment by NGSpace** [comment]
As for the not showing up you can use the ;mute; method
